src/wayland/session_lock/mod.rs

```diff
@@ -0,0 +1,220 @@
+//! Utilities for handling the `ext-session-lock` protocol
+//!
+//! ## How to use it
+//!
+//! ### Initialization
+//!
+//! To initialize this implementation create the [`SessionLockManagerState`] and
+//! implement the [`SessionLockHandler`], as shown in this example:
+//!
+//! ```
+//! use smithay::delegate_session_lock;
+//! use smithay::reexports::wayland_server::protocol::wl_output::WlOutput;
+//! use smithay::wayland::session_lock::{SessionLockManagerState, SessionLockHandler, SessionLocker};
+//! use smithay::wayland::session_lock::surface::LockSurface;
+//!
+//! # struct State { session_lock_state: SessionLockManagerState }
+//! # let mut display = wayland_server::Display::<State>::new().unwrap();
+//! // Create the compositor state
+//! let session_lock_state = SessionLockManagerState::new::<State>(&display.handle());
+//!
+//! // Insert the SessionLockManagerState into your state.
+//!
+//! // Implement the necessary trait.
+//! impl SessionLockHandler for State {
+//!     fn lock_state(&mut self) -> &mut SessionLockManagerState {
+//!         &mut self.session_lock_state
+//!     }
+//!
+//!     fn lock(&mut self, _confirmation: SessionLocker) {
+//!         // Lock and clear the screen.
+//!
+//!         // Call `SessionLocker::lock` after the cleared frame was presented.
+//!     }
+//!
+//!     fn unlock(&mut self) {
+//!         // Remove session lock.
+//!     }
+//!
+//!     fn new_surface(&mut self, _surface: LockSurface, _output: WlOutput) {
+//!         // Display `LockSurface` on `WlOutput`.
+//!     }
+//! }
+//! delegate_session_lock!(State);
+//!
+//! // You're now ready to go!
+//! ```
+
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+
+use _session_lock::ext_session_lock_manager_v1::{ExtSessionLockManagerV1, Request};
+use _session_lock::ext_session_lock_v1::ExtSessionLockV1;
+use wayland_protocols::ext::session_lock::v1::server as _session_lock;
+use wayland_server::protocol::wl_output::WlOutput;
+use wayland_server::protocol::wl_surface::WlSurface;
+use wayland_server::{Client, DataInit, Dispatch, DisplayHandle, GlobalDispatch, New};
+
+use crate::wayland::session_lock::lock::SessionLockState;
+use crate::wayland::session_lock::surface::{LockSurface, LockSurfaceConfigure};
+
+pub mod lock;
+pub mod surface;
```

Comment:
I think for other protocols in Smithay, modules like this are private, and their types are re-exported.